doc/user_quide.asciidoc

```diff
@@ -0,0 +1,105 @@
+# SUSE Distribution Migration System
+===========
+:Authors: Marcus Schäfer and Jesús Bermúdez Velázquez
+:Publication_Date: TBD
+:Latest_Version: 0.5.15
+:Contributors: 
+:Repo: https://github.com/SUSE/suse-migration-services[suse-migration-services]
+
+[%hardbreaks]
+Authors: {Authors}
+Contributors: {Contributors}
+Publication Date: {Publication_Date}
+Latest Version: {Latest_Version}
+Code available: {Repo}
+
+=== Concept
+Upgrading from one release to a newer one having to skip a range of multiple
+releases introduces an upgrade problem.
+
+The Distribution Migration System provides an upgrade path from A to B,
+where A and B are to major distributions versions (for example, from SLES 12 SP4 to SLES 15 GA).
+
+=== Requirements
+The system to be migrated must be registered.
+
+[listing]
+$ SUSEConnect --regcod [SLES_REGCODE]
+
+=== How does distribution migration system do it ?
+To achieve the migration between this two releases, the entire migration process
+is called from within a Live Migration Image.
+
+Once the Distribution Migration system package has been installed in
+the system to be migrated and it starts running, a pipeline of systemd
+controlled services start to run. The following chain of services will be executed:
+
+- detect the system to migrate
+- mount it
+- setup the network as clone from the system to migrate
+- prepare the package manager for the migration task
+- migrate the system
+- update the grub
+- unmount what it mounted
+- reboot
+
+=== Installation
+
+Inside the system to be migrated the following tasks have to be performed:
+
+. fetch the Distribution Migration System package which contains a
+KIWI built image based on the target distribution of the migration:
++
+[listing]
+$ zypper ar official_repo_url live_migration
+
+. fetch the suse-migration-activation package which contains an
+activation package:
++
+[listing]
+$ zypper ar official_repo_url suse_migration_activation
+
+. install it:
++
+[listing]
+$ zypper in SLE15-Migration suse_migration_activation
+
+=== Run migration
+
+When the package has been installed, run:
+
+[listing]
+$ reboot
+
+When the migration has started, the only way to access the system during the
```

Comment:
Hm - that means that we can not login and debug in case the network setup does not work or in case there is a problem with the SSH keys or SSH configuration (what ever that might be).

Could we also login on the console with the migration or any other user? 
Or could we use the passwd of the system to be upgraded to allow user-login to the console while the upgrade is running? (i.e. in case the system we upgrade does not have any authorized keys)

Reply:
> Hm - that means that we can not login and debug in case the network setup does not work or in 
> case there is a problem with the SSH keys or SSH configuration (what ever that might be).

correct. So this process is made for "no physical access to the machine possible". This means to install the migration packages some access method to the system which is not "a human being sits on the computer" must exist. Access to a machine with a password is considered insecure and so we don't allow that and require ssh.

* if the network could not be setup, login is the smallest problem ;) you will be returned to the former state without any changes. Given we have sorted out all the network dependency issues there is not much that should prevent a former working network to run again in the live migration system.

* if the access method to the instance was password based you will be on a black period until the migration is done.

Reply:
> Could we also login on the console with the migration or any other user?

No during migration only the migration user exists. This is on purpose to make clear the system is just there for running the migration not for working with it

Reply:
> Or could we use the passwd of the system...

Nope sorry. The image as we deliver does not contain any credentials. The migration user is restricted to ssh based auth_key login. All for security reasons

Reply:
We also don't take over any credentials from the system that gets migrated into the live migration system. Only public keys from the authorized_keys files from the existing users are taken into account. Again all for security reasons

Reply:
Ok - then I think we really should advice the customer to add at least one SSH key to the authorized_keys file. 

And maybe we also should add an "unsecure" option that allows debugging of network problems on the servers console.

Reply:
> Ok - then I think we really should advice the customer to add at least one SSH key

yes as a preparation task if not present I agree that would be good

> And maybe we also should add an "unsecure" option that allows debugging 

You mean access by a simple password that a user could set in the customer config file ? might be possible. If you like you can add that as an enhancement issue here

Reply:
Robert added a recommendation about ssh usage